subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/FetchReleaseNotes.java

```diff
@@ -16,6 +18,15 @@
 
 public class FetchReleaseNotes {
     public void fetchReleaseNotes(FetchReleaseNotesTask task) {
+        try {
+            performFetchReleaseNotes(task);
```

Comment:
Can you try using the ExceptionHandling class we recently added? This way, you can create a clean unit test for that complexity.

Reply:
If we want to use ExceptionHandling class in this case, we need to change its API from 
```public static void withExceptionHandling(Runnable code, Action<Exception> handler) ```
to  
```public static void withExceptionHandling(Runnable code, Action<Exception> handler) ```

There's another solution to make ```RevisionNotFoundException``` unchecked.

Reply:
If that's the case, let's move on without using ExceptionHandling class. Sorry for late feedback! I didn't realize that we're blocked on this PR.

Bouns: you can try to write a unit test using spock's "spy" feature that enables to mock one method from a class. We could use it to simulate "performFetchReleaseNotes" method to throw an exception.

Reply:
OK. I left that, because I could not mock ``performFetchReleaseNotes``.